build.psm1

```diff
@@ -516,11 +516,8 @@ Fix steps:
 
     # setup arguments
     $Arguments = @("publish","--no-restore","/property:GenerateFullPaths=true")
-    if ($Output) {
-        $Arguments += "--output", $Output
-    }
-    elseif ($SMAOnly) {
-        $Arguments += "--output", (Split-Path $script:Options.Output)
+    if ($Output -or $SMAOnly) {
```

Comment:
The -Output parameter and the Output property in $PSOptions is two different items.
`Start-PSBuild -Output`/`New-PSOptions -Output` accepts a path to the output directory.
New-PSObject appends the pwsh binary name to a directory path, whether -Output is specified or not, and assign it to $PSOptions.Output.

Reply:
What I see is the following:
New-PSOptions 
If $Output from Start-PSBuild is null or empty; it is updated with the expected pwsh.exe path; otherwise, it returns the passed in value unchanged.

Start-PSBuild
If $Output is specified as a directory path to Start-PSBuild, --output will be set to the parent directory.

Reply:
>New-PSOptions
If $Output from Start-PSBuild is null or empty; it is updated with the expected pwsh.exe path; otherwise, it returns the passed in value unchanged.

That is how the original code handled -Output. Please find line 817-818 in the commit.
https://github.com/PowerShell/PowerShell/blob/69a07f3b13481a388490d7c34d3305d41c0bf314/build.psm1#L814-L819

>Start-PSBuild
If $Output is specified as a directory path to Start-PSBuild, --output will be set to the parent directory.

Yes. If -Output is specified, --output will be set to the parent directory of $Options.Output, which contains full path to the pwsh.exe.

Reply:
@dantraMSFT thoughts on this?

Reply:
@dantraMSFT ping.